Added start of dashboard-sample for site (#843)
fybrik/fybrik: site/docs/samples/fog-protect.md

# Dashboard sample

> This use-case demonstrates how Fybrik uses runtime evaluation of policies 
> to control access privileges for multiple assets.

Reviewer: ```suggestion
This sample show how Fybrik enables a dashboard application to display data from a backend REST server and displaying only compliant information with the person operating the dashboard.
In the sample we show how a Fybrik module that support REST protocol can control what information is displayed in the dashboard through policies.

Specifically in this sample we demonstrate:
- Just in time policy decisions 
- a Fybrik module that supports REST API 

Note: At this time some features in this demo use mocks as some underlying support in Fybrik is still pending.
```

> ## Mockup version of the real scenario

Reviewer: ```suggestion
## About the dashboard application
```

> This is based on the smart manufacturing scenario of *Fogprotect*, 
> where there exists a manufacturing factory which can be controlled/supervised via 
> transmitted commands, without the need of the staff member to be physically present 
> in the factory.  

Reviewer: ```suggestion
The dashboard application here is based on the smart manufacturing use-case of the European Horizon 2020 research project [Fogprotect](https://fogprotect.eu/). Details for the use case are available [here](https://fogprotect.eu/results/#use-cases).
The dashboard included is this sample is a mock dashboard of a remote manufacturing factory which can be controlled/supervised by expert operators from remote, without the need for the expert operator to be physically present in the factory.  
```

> The scenario here is simplified for the sake of the demo. The resources in this demo are:  

Reviewer: ```suggestion
The scenario here is simplified for the purpose of this sample and demonstrates the following scenario:
```

> 1. A robot in the manufacturing area that can be started/stopped.  
> 2. Two factory sectors, a production and a non-production sector, where some employees wear helmets 
> and some don't. There's a video camera with image processing which monitors and counts the number of employees 
> wearing/not wearing helmets in each sector. This is called the safety data of the factory.  

Reviewer: ```suggestion
1. A manufacturing robot that operates the manufacturing facility that can be activated or stopped remotely.
2. A safety dashboard observing the manufacturing floor which is composed of two areas. A production and a non-production areas, which are observed by a video camera with image processing which monitors and counts the number of employees with and without helmets in each area. This is called the safety data of the factory.  
```

> There are multiple roles in this scenario:  

Reviewer: ```suggestion
We demonstrate Fybrik abilities to control what data is seen by each role. In this scenario we show the following roles:  
```

> 1. A *Foreman* who is allowed to access all of the assets because he is the foreman of the factory. 

Reviewer: ```suggestion
1. A *Foreman* who is allowed to access all of the assets. 
```

Furthermore, only the Foreman is allowed to control the robot in the manufacturing area.  
2. A *Worker* who is not allowed to control the robot, and doesn't have privileges to see the number 
of employees wearing/not wearing helmets in each of the available sectors. However, a Worker can see the total 

> number of employees in each sector.  

Reviewer: change sector ---> area

> 3. An *HR* who is also not allowed to control the robot, but has access to view the number of employees 

Reviewer: ```suggestion
3. *HR personnel* who is also not allowed to control the robot, but has access to view the number of employees 
```

wearing/not wearing helmets in each sector.

> ## Architecture

Reviewer: ```suggestion
## Dashboard sample architecture
```

The project contains 3 main components:  

> - A backend data server, responsible for reading/writing data (possibly to a database).  

Reviewer: is this the backend data service? if so, just say it provides the mock data for the dashboard.

> - A proxy server, responsible for intercepting HTTP requests sent

Reviewer: is this the fybrik module? if so, please describe it as a module and provide a reference to the github where the module reside.

from a user trying to read/write data.  

> - A frontend GUI/dashboard, which helps the user perform the HTTP 

Reviewer: this is the dashboard application, right? if so, please describe it as an application.

requests.

For a more detailed description of the implementation visit [fogProtect-dashboard-sample](https://github.com/fybrik/fogProtect-dashboard-sample/tree/chart-push).

## Before you begin

- Install Fybrik using the [Quick Start](../get-started/quickstart.md) guide.
- A web browser.

## Create a namespace for the sample

Create a new Kubernetes namespace and set it as the active namespace:

```bash
kubectl create namespace fogprotect
kubectl config set-context --current --namespace=fogprotect
```

This enables easy [cleanup](#cleanup) once you're done experimenting with the sample.

## Backend data server

The backend data server is responsible for reading/writing data to a database or any other resource. In this 
example the backend data server is a simple server that returns random numbers whenever a read request is 

> received.  

Reviewer: you can just say it returns mock data, no need for details.
